### PR TITLE
PublicIPAddress support

### PR DESCRIPTION
**Pull request by vaspahomov**
Hello! 

I want to add VM instances support in crossplane/provider-azure. 

This is first part of big patch with Public IP addresses support

### Description of your changes

Add support of Public IP Address.
Fixes https://github.com/crossplane/provider-azure/issues/208

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Same like in other resources like Subnet and VisrtualNetwork. And also manually tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by ulucinar — CHANGES_REQUESTED** [review]

**Review by ulucinar — COMMENTED** [review]
Thank you @vaspahomov for your work on this PR! Could you please rebase this PR and make sure all your commits are signed following the guide [here](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/signing-commits). We also require them to be signed-off using something similar to `git commit -s`. `git log --show-signature` should display commit signatures.

File: pkg/controller/network/publicipaddress/managed.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	_, ok := mg.(*v1alpha3.PublicIPAddress)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotPublicIPAddress)
+	}
```
Comment (ulucinar) [inline pkg/controller/network/publicipaddress/managed.go:133]:
```suggestion
```

File: pkg/controller/network/publicipaddress/managed.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+	network.UpdatePublicIPAddressStatusFromAzure(s, az)
+	s.SetConditions(xpv1.Available())
+
+	o := managed.ExternalObservation{
+		ResourceExists:    true,
+		ConnectionDetails: managed.ConnectionDetails{},
+	}
+
+	return o, nil
```
[the remark this answers is not in the document]
Comment (vaspahomov) [reply pkg/controller/network/publicipaddress/managed.go]: I've add todo about update support. It's not very often case for now

**Comment by vaspahomov** [comment]
Hi @ulucinar thank you! I've commit fixes

**Comment by ulucinar** [comment]
Hi @vaspahomov,
Thank you for working on this PR. I'm planning to merge this and maybe address some remaining comments in a follow-up PR? What do you think?

**Review by ulucinar — APPROVED** [review]

**Comment by ulucinar** [comment]
Thank you very much @vaspahomov, merging this PR.

**Comment by ulucinar** [comment]
@vaspahomov, I've just opened a follow-up [PR](https://github.com/crossplane/provider-azure/pull/309).

**Pull request merged**